### PR TITLE
fix: configure tasks from open task quick pick

### LINKED ISSUE #13086: Cannot configure provided tasks
### Bug Description:

In a Theia application that has VS Code extensions providing tasks (e.g. for npm, grunt, etc.), they are correctly shown when executing *Run Task...*. However, when I select a provided task folder (e.g. npm), I cannot click on the cogwheel to configure them in my `tasks.json`.

### Steps to Reproduce:

1. Open Blueprint (or any other Theia application with VS Code extension that contribute provided tasks)
2. Open a workspace where a provided task is discoverable (e.g. a folder with an npm module, like a Theia application)
3. Select *npm*
4. Try to click on the cogwheel

https://github.com/eclipse-theia/theia/assets/588090/3bfd4ca3-1c27-4ae5-bda8-ae80960b2956

What happens: nothing
What should happen: the provided task is copied over to tasks.json for further customization.

### Additional Information

- Operating System: Ubuntu Linux
- Theia Version: 1.43.0

### PR DESCRIPTION
**Pull request by AlexandraBuzila**
#### What it does

The gear icon in the task quick pick should open the task configuration.

Fixes #13086

Contributed on behalf of STMicroelectronics

#### How to test

1. Open the task quick pick from _>Task: Run Task..._
2. For any configured task, press the gear icon  → the `tasks.json` configuration file should open

Or:
1. Open the task quick pick from _>Task: Run Task..._
2. Select a task category from the dialog, e.g. `npm`
4. In the list of tasks, press the gear icon → the `tasks.json` configuration file should open

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/task/src/browser/quick-open-task.ts [hunk 1]
```diff
@@ -182,6 +182,7 @@ export class QuickOpenTask implements QuickAccessProvider {
         picker.matchOnDescription = true;
         picker.ignoreFocusOut = false;
         picker.items = this.items;
+        picker.onDidTriggerItemButton(({ item }) => this.onDidTriggerGearIcon(item as QuickPickItem));
```
Comment (tsmaeder) [inline packages/task/src/browser/quick-open-task.ts:185]: It seems odd that we have to case here, but not below. Any idea why?

Comment (AlexandraBuzila) [reply packages/task/src/browser/quick-open-task.ts]: The two methods have different signatures :smile: 
Here the item can be a separator as well, even if it won't happen (separators  won't have buttons that will trigger this method). However, to make this nicer I updated the code and added a check here, instead.

Comment (tsmaeder) [reply packages/task/src/browser/quick-open-task.ts]: That was the question: is it true that one time the button item can be a separator as well or is that a mistake in the typing of "picker.onDidTriggerItemButton()"?

Comment (AlexandraBuzila) [reply packages/task/src/browser/quick-open-task.ts]: Hi @tsmaeder! No, the item passed to the button cannot be a separator. The separator only has a label and no buttons. I restricted the type in the interface for this method. Ideally, we would have a dedicated type for a separator picker, that does not have this method at all, but I think that refactoring is out of scope for this PR.

**Review by tsmaeder — APPROVED** [review]
:lgtm:

**Pull request merged**